Realm/Realm/Sync/ErrorHandling/DiscardLocalResetHandler.cs

```diff
@@ -0,0 +1,67 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+namespace Realms.Sync.ErrorHandling
+{
+    /// <summary>
+    /// A client reset strategy where all the not yet synchronized data is automatically discarded and a fresh copy of the synchronized Realm is obtained.
+    /// To be noted that the freshly downloaded copy of the synchronized Realm triggers all change notifications as a write transaction is internally simulated.
+    /// This strategy supplies three callbacks: <see cref="OnBeforeReset"/> and <see cref="OnAfterReset"/> to give a chance to take some user actions,
```

Comment:
The third is at line 26

Reply:
Ah, I think the wording tricked me here - I think we should list all three callbacks in a single sentence - something like:

```
This strategy supplies three callbacks: <see cref="OnBeforeReset"/>, <see cref="OnAfterReset"/>, and <see cref="ManualResetFallback"/>. The first two give notify you just before and after the client reset has happened, while the last one will be invoked in case an error occurs during the automated process and the system needs to fallback to a manual mode.
```